api/tests/unit/domain/strategies/SmartRandom_test.js

```diff
@@ -488,5 +488,37 @@ describe('Unit | Domain | Models | SmartRandom', () => {
       // then
       expect(result).to.deep.equal([challengeAssessingSkill3]);
     });
+
+    context('when a challenge than validated skill not in target profil has been answered', () => {
```

Comment:
context("when the selected challenges covers more skills than the defined target profile")
it('should ignore a challenge that already answered, even if it has non evaluated skills', function() {